lib/std/Build/Module.zig

```diff
@@ -557,6 +559,10 @@ pub fn appendZigProcessFlags(
     try addFlag(zig_args, m.pic, "-fPIC", "-fno-PIC");
     try addFlag(zig_args, m.red_zone, "-mred-zone", "-mno-red-zone");
 
+    if (m.root_source_file != null) {
+        try addFlag(zig_args, m.allow_deprecated, "-fallow-deprecated", "-fno-allow-deprecated");
```

Comment:
If it's desirable to prevent individual modules in the Zig build system from overriding their allow-deprecated behavior, here we could instead check the `is_root` property of the builder that owns `m`. 

As mentioned earlier this would not be a full solution though, since the root builder could be modified by the user.

Reply:
I think this is a better idea. It's not so much that we want to stop people from sneakily changing the field; it's more that this is just a simpler way to do it (don't duplicate data needlessly!).

I definitely don't think the build system should *support* changing this flag on a per-module basis for now; assume YAGNI.